src/slice-chop.c

```diff
@@ -296,6 +296,10 @@ static SEXP chop_shaped(SEXP x, SEXP indices, struct vctrs_chop_info info) {
 }
 
 static SEXP chop_fallback(SEXP x, SEXP indices, struct vctrs_chop_info info) {
+  // TODO: Do we really care about micro performance that much here? Can we just
+  // merge with `chop_shaped_fallback()` now that `vec_slice_fallback()`
+  // handles shaped an unshaped vectors?
```

Comment:
This `chop_fallback()` only runs on S3/S4 objects that don't have a `vec_proxy()` method defined for them (i.e. `vec_requires_fallback()` is `true`).

I can't imagine we care about the performance of those cases _that_ much. I think I'd rather simplify this code by a lot by merging it with `chop_fallback_shaped()`, which just allows `vec_slice_fallback()` to handle everything related to setting up the environment and doing `vec_is_restored()`. The less places we have that `vec_is_restored()` check lying around, the better IMO.